### PR TITLE
chore: deprecate constants

### PR DESCRIPTION
**Pull request by majorgreys**
Deprecate constants that are intended for internal usage only.

### CODE REVIEW

File: tests/tracer/test_constants.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import warnings
+
+import pytest
+
+
+def test_deprecated_constants():
```
Comment (brettlangdon) [inline tests/tracer/test_constants.py:6]: should we assert that non-deprecated don't show warning?

File: ddtrace/constants.py [hunk 2]
```diff
@@ -36,3 +37,30 @@
 AUTO_KEEP = 1
 # Use this to explicitly inform the backend that a trace should be kept and stored.
 USER_KEEP = 2
+
+
+_DEPRECATED = {
+    "FILTERS_KEY": "FILTERS",
```
Comment (brettlangdon) [inline ddtrace/constants.py:43]: maybe a comment here saying that we are removing this entirely as opposed to moving.

it took me a second of reviewing the below TODO comment why this was different.

**Review by brettlangdon — COMMENTED** [review]
minor notes, but lgtm

File: ddtrace/constants.py (hunk 2 above)
Comment (majorgreys) [reply ddtrace/constants.py:46]: Missed in commit, fixed.

**Review by brettlangdon — DISMISSED** [review]

**Review by Kyle-Verhoog — DISMISSED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by mabdinur — APPROVED** [review]

**Pull request merged**